flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1842,14 +1869,18 @@ private void closeJob(JobTable.Job job, Exception cause) {
         job.asConnection()
                 .ifPresent(
                         jobManagerConnection ->
-                                disconnectJobManagerConnection(jobManagerConnection, cause));
+                                disconnectJobManagerConnection(jobManagerConnection, cause, false));
 
         job.close();
     }
 
     private void disconnectJobManagerConnection(
-            JobTable.Connection jobManagerConnection, Exception cause) {
+            JobTable.Connection jobManagerConnection,
+            Exception cause,
+            boolean possibleRetainPartitions) {
```

Comment:
Maybe rename `possibleRetainPartitions` to `releasePartitions` and reverse all the current checks and value settings.